backend/src/main/java/ai/verta/modeldb/versioning/FindRepositoriesQuery.java

```diff
@@ -0,0 +1,355 @@
+                    .append(alias)
+                    .append(".")
+                    .append(ModelDBConstants.ID)
+                    .append(" AND ");
+                joinClause
+                    .append(ModelDBConstants.LABELS)
+                    .append(".id.entity_type = ")
+                    .append(IDTypeEnum.IDType.VERSIONING_REPOSITORY.getNumber());
+              }
+            });
+      }
+
+      StringBuilder whereClause = new StringBuilder();
+      if (workspaceDTO != null
+          && workspaceDTO.getWorkspaceId() != null
+          && !workspaceDTO.getWorkspaceId().isEmpty()) {
+        whereClause.append(" WHERE ");
+        whereClause
+            .append(alias)
+            .append(".")
+            .append(ModelDBConstants.WORKSPACE_ID)
+            .append(" = :")
+            .append(ModelDBConstants.WORKSPACE_ID);
+        parametersMap.put(ModelDBConstants.WORKSPACE_ID, workspaceDTO.getWorkspaceId());
+        whereClause
+            .append(" AND ")
+            .append(alias)
+            .append(".")
+            .append(ModelDBConstants.WORKSPACE_TYPE)
+            .append(" = :")
+            .append(ModelDBConstants.WORKSPACE_TYPE);
+        parametersMap.put(
+            ModelDBConstants.WORKSPACE_TYPE, workspaceDTO.getWorkspaceType().getNumber());
+        isNeedOperator = true;
+      } else if (this.isNeedToWhereCause) {
+        whereClause.append(" WHERE ");
+      }
+
+      if (this.predicates != null && !this.predicates.isEmpty()) {
+        if (isNeedOperator) {
+          whereClause.append(" AND ");
+        }
+        for (int index = 0; index < this.predicates.size(); index++) {
+          KeyValueQuery keyValueQuery = this.predicates.get(index);
+          if (keyValueQuery.getKey().contains(ModelDBConstants.LABEL)) {
+            whereClause.append(ModelDBConstants.LABELS).append(".id.label ");
+            setPredicateInQuery(whereClause, keyValueQuery);
+          } else {
+            whereClause.append(alias).append(".").append(keyValueQuery.getKey());
+            setPredicateInQuery(whereClause, keyValueQuery);
+          }
+          if (index < this.predicates.size() - 1) {
+            whereClause.append(" AND ");
+          }
+        }
+        isNeedOperator = true;
+      }
+
+      if (this.repoIds != null && !this.repoIds.isEmpty()) {
+        if (isNeedOperator) {
+          whereClause.append(" AND ");
+        }
+        whereClause.append(alias).append(".").append(ModelDBConstants.ID).append(" IN (:repoIds) ");
+        parametersMap.put("repoIds", this.repoIds);
+        isNeedOperator = true;
+      }
+
+      // Order by clause
+      StringBuilder orderClause =
+          new StringBuilder(" ORDER BY")
+              .append(alias)
+              .append(".")
+              .append(ModelDBConstants.DATE_UPDATED)
+              .append(" DESC");
+
+      StringBuilder finalQueryBuilder = new StringBuilder();
+      if (!joinClause.toString().isEmpty()) {
```

Comment:
what happens if there is no join and we add a select? is it not recommended or is outright incorrect and throw s error?

Reply:
if it still works then i would suggest removing the if

Reply:
- If there is a join then hibernate returns the two entities. I mean root & join entity both so at that time we should have to pass select clause like `finalQueryBuilder.append("SELECT ").append(alias).append(" ");` that means root entity so it will return only a single root entity.
- If there is not a join then in hibernate not need to add Select clause so we build a query from `new StringBuilder(" FROM ").append(RepositoryEntity.class.getSimpleName()).append(alias)` FROM clause so it will return root entity.
So above this is needed in code.